rust/processor/src/bq_analytics/generic_parquet_processor.rs

```diff
@@ -153,82 +164,102 @@ where
-        if self.buffer_size_bytes >= max_buffer_size {
-            let start_version = self.buffer.first().unwrap().version();
-            let last = self.buffer.last().unwrap();
-            let end_version = last.version();
-            let last_transaction_timestamp = naive_datetime_to_timestamp(last.get_timestamp());
-
-            let txn_version_to_struct_count = process_struct_count_map(
-                &self.buffer,
-                &mut self.transaction_version_to_struct_count,
-            );
-
-            let new_file_path: PathBuf = PathBuf::from(format!(
-                "{}_{}.parquet",
-                ParquetType::TABLE_NAME,
-                Uuid::new_v4()
-            ));
-            rename(&self.file_path, &new_file_path)?; // this fixes an issue with concurrent file access issues
-
-            let struct_buffer = std::mem::take(&mut self.buffer);
-
-            let mut row_group_writer = self.writer.next_row_group()?;
-            struct_buffer
-                .as_slice()
-                .write_to_row_group(&mut row_group_writer)
-                .unwrap();
-            row_group_writer.close()?;
-            self.close_writer()?;
-
-            debug!(
-                table_name = ParquetType::TABLE_NAME,
-                start_version = start_version,
-                end_version = end_version,
-                "Max buffer size reached, uploading to GCS."
-            );
-            let bucket_root = PathBuf::from(&self.bucket_root);
-            let upload_result = upload_parquet_to_gcs(
-                gcs_client,
-                &new_file_path,
-                ParquetType::TABLE_NAME,
-                &self.bucket_name,
-                &bucket_root,
-            )
-            .await;
-            self.buffer_size_bytes = 0;
-            remove_file(&new_file_path)?;
-
-            return match upload_result {
-                Ok(_) => {
-                    let parquet_processing_result = ParquetProcessingResult {
-                        start_version,
-                        end_version,
-                        last_transaction_timestamp: Some(last_transaction_timestamp),
-                        txn_version_to_struct_count,
-                    };
-
-                    self.gap_detector_sender
-                        .send(ProcessingResult::ParquetProcessingResult(
-                            parquet_processing_result,
-                        ))
-                        .await
-                        .expect("[Parser] Failed to send versions to gap detector");
-                    Ok(())
-                },
-                Err(e) => {
-                    error!("Failed to upload file to GCS: {}", e);
-                    Err(anyhow!("Failed to upload file to GCS: {}", e))
-                },
-            };
+            if self.buffer_size_bytes >= self.max_buffer_size {
+                if let Err(e) = self.upload_buffer(gcs_client).await {
```

Comment:
nit: you can add a different log here to say buffer size large and we upload; this is for easily debugging.